sopel/modules/currency.py

```diff
@@ -2,109 +2,209 @@
+    # Update fiat rates
+    if bot.config.currency.fixer_io_key is not None:
+        response = requests.get(FIXER_URL.format(bot.config.currency.fixer_io_key))
+        if not response.json()['success']:
+            raise FixerError('Fixer.io request failed with error: {}'.format(response.json()['error']))
+    else:
+        response = requests.get(FIAT_URL)
+
+    response.raise_for_status()
+    rates_fiat_json = response.json()
+    rates_updated = time.time()
+    rates_fiat_json['rates']['EUR'] = 1.0  # Put this here to make logic easier
+
+
+def btc_rate(code, reverse=False):
+    search = 'BTC{}'.format(code)
+
+    if search in rates_btc_json:
+        rate = rates_btc_json[search]['averages']['day']
+    else:
+        return UNSUPPORTED_CURRENCY.format(code)
+
+    if reverse:
+        return 1 / rate
+    else:
+        return rate
+
+
+def get_rate(of, to):
+    of = of.upper()
+    to = to.upper()
+
+    if of == 'BTC':
+        return btc_rate(to, False)
+    elif to == 'BTC':
+        return btc_rate(of, True)
+
+    if of not in rates_fiat_json['rates']:
+        return UNSUPPORTED_CURRENCY.format(of)
+
+    if to not in rates_fiat_json['rates']:
+        return UNSUPPORTED_CURRENCY.format(to)
+
+    return (1 / rates_fiat_json['rates'][of]) * rates_fiat_json['rates'][to]
+
+
+def exchange(bot, match):
     """Show the exchange rate between two currencies"""
-    if not trigger.group(2):
-        return bot.reply("No search term. An example: .cur 20 EUR in USD")
-    match = regex.match(trigger.group(2))
     if not match:
-        # It's apologetic, because it's using Canadian data.
-        bot.reply("Sorry, I didn't understand the input.")
+        bot.reply(UNRECOGNIZED_INPUT)
+        return NOLIMIT
+
+    try:
+        update_rates(bot)  # Try and update rates. Rate-limiting is done in update_rates()
+    except requests.exceptions.RequestException as err:
+        bot.reply("Something went wrong while I was getting the exchange rate.")
+        LOGGER.error("Error in GET request: {}".format(err))
+        return NOLIMIT
+    except ValueError:
+        bot.reply("Error: Got malformed data.")
+        LOGGER.error("Invalid json on update_rates")
         return NOLIMIT
+    except FixerError as err:
+        bot.reply('Sorry, something went wrong with fixer')
+        LOGGER.error(err)
+        return NOLIMIT
+
+    query = match.string
+
+    others = query.split()
+    amount = others.pop(0)
+    of = others.pop(0)
+    others.pop(0)
+
+    # TODO: Use this instead after dropping Python 2 support
+    # amount, of, _, *others = query.split()
 
-    amount, of, to = match.groups()
     try:
         amount = float(amount)
     except ValueError:
-        bot.reply("Sorry, I didn't understand the input.")
+        bot.reply(UNRECOGNIZED_INPUT)
     except OverflowError:
         bot.reply("Sorry, input amount was out of range.")
-    display(bot, amount, of, to)
 
+    out_string = '{} {} is'.format(amount, of.upper())
```

Comment:
Always printing the input with two decimal places would look better.

Reply:
Unless the input currency doesn't use decimals, like JPY.